packages/block-library/src/navigation/index.php

```diff
@@ -793,35 +817,103 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 	$toggle_aria_label_open      = $should_display_icon_label ? 'aria-label="' . __( 'Open menu' ) . '"' : ''; // Open button label.
 	$toggle_aria_label_close     = $should_display_icon_label ? 'aria-label="' . __( 'Close menu' ) . '"' : ''; // Close button label.
 
-	$responsive_container_markup = sprintf(
-		'<button aria-haspopup="true" %3$s class="%6$s" data-micromodal-trigger="%1$s">%9$s</button>
-			<div class="%5$s" style="%7$s" id="%1$s">
-				<div class="wp-block-navigation__responsive-close" tabindex="-1" data-micromodal-close>
-					<div class="wp-block-navigation__responsive-dialog" aria-label="%8$s">
-							<button %4$s data-micromodal-close class="wp-block-navigation__responsive-container-close">%10$s</button>
+	if ( $is_interactivity_api_enabled ) {
```

Comment:
Would it be possible to use the tag processor instead to inject directives necessary for Interactivity API? This way we would need only a single markup definition for both, and the rest would be conditionally applied when the Interactivity API is enabled. I might be missing something that validates having two copies though.

Reply:
I can definitely try that. It's something I thought but I wasn't sure what is better:
* Having the markup with the directives as it will be once/if it is not experimental anymore. I feel it is easier to understand it.
* Using the tag processor to avoid code duplication.
As I say, I don't have a strong opinion and I am not sure which one is better. I'd love to hear your thoughts.

Reply:
Let's ignore my comment for now. It isn't simple to compare what changes between the new and the old version, but it's also a temporary approach for the testing period where we plan to validate the new approach. In a few weeks, we will most likely remove the old version and this way the process will be straightforward.

Reply:
I've just pushed some commits to implement it using the tag processor, as you suggested, to be able to compare. It certainly looks cleaner, and we don't have to modify so many blocks.

**With the tag processor**

* [Navigation block](https://github.com/WordPress/gutenberg/blob/1d0472db69a1d2fae021f6591a5eb7163318dab2/packages/block-library/src/navigation/index.php): We just have to add the scripts conditionally (maybe this can be moved as well).
* [Navigation submenu block](https://github.com/WordPress/gutenberg/blob/1d0472db69a1d2fae021f6591a5eb7163318dab2/packages/block-library/src/navigation-submenu/index.php): It doesn't change at all.
* [Page list block](https://github.com/WordPress/gutenberg/blob/1d0472db69a1d2fae021f6591a5eb7163318dab2/packages/block-library/src/page-list/index.php): It doesn't change at all.

The benefits I see are that it looks much simpler and it might be more compatible with other changes made to the navigation block.

**Without the tag processor**

* [Navigation block](https://github.com/WordPress/gutenberg/blob/e63c002f0a8a596ee587f94f6d905d218d0785ed/packages/block-library/src/navigation/index.php): Apart from adding the scripts conditionally, we have to include more conditionals to change the markup, and duplicate it in most parts.
* [Navigation submenu block](https://github.com/WordPress/gutenberg/blob/e63c002f0a8a596ee587f94f6d905d218d0785ed/packages/block-library/src/navigation-submenu/index.php): We have to include conditionals to change the markup, and duplicate it.
* [Page list block](https://github.com/WordPress/gutenberg/blob/e63c002f0a8a596ee587f94f6d905d218d0785ed/packages/block-library/src/page-list/index.php): We have to include conditionals to change the markup, and duplicate it.

The benefits I see are that it is clearer how to use directives and the old markup should be removed once/if it is not experimental anymore.

Reply:
I responded in the comment with general feedback in favor of using the tag processor: https://github.com/WordPress/gutenberg/pull/50041#issuecomment-1527427568.